packages/flutter/lib/src/painting/image_provider.dart

```diff
@@ -270,10 +270,19 @@ abstract class ImageProvider<T> {
   /// This is the public entry-point of the [ImageProvider] class hierarchy.
```

Comment:
please add a section to the class docs (`## foo`) that discusses the lifecycle of this class and the order in which the various methods are called, by whom, when, and so on. The class is complicated enough now that an overview is necessary.

Reply:
+1